### DISCUSSION TITLE
Auto-completion not (always) available with functions with decorators from a library

### DISCUSSION THREAD

**Discussion by leana8959** [microsoft/pyright#6565] [Q&A] [answered]
Hello,

When I use `pyright` with `discord.py`, I noticed that the functions' parameters aren't typed, and this prevents the auto-completion from working. Is there a solution to this, or is this a known limitation ?

Here's an example:
<img width="447" alt="image" src="https://github.com/microsoft/pyright/assets/87855546/3af3d200-88a2-425d-a95b-831ae77306cb">

Unable to deduce possible methods without manual typing.

<img width="731" alt="image" src="https://github.com/microsoft/pyright/assets/87855546/2cdc0dda-e693-4327-b6f2-ef566bf09fb4">

Auto-completion is possible only with manual typing in the arguments.

Thanks for reading my question, and for the time developing this project :)

**Comment by erictraut** [comment, marked as answer]
It's difficult to diagnose issues from partial screen shots. From what I can discern in your top screen shot, you haven't provided a type annotation for the input parameter `message`, so pyright has no idea what type this symbol is. In the bottom screen shot, you've provided the requisite type annotation for the `message` parameter. Unless I'm missing something, this has nothing to do with decorators or library methods. It simply has to do with the fact that type annotations are required for input parameters if you want pyright to have any sense of its type. Without a type annotation, it will infer a type of `Any`, so no completion information will appear.
